model/project_parser.go

```diff
@@ -644,40 +697,64 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 		return nil, errors.Wrapf(err, LoadProjectError)
 	}
 
-	// return intermediateProject even if we run into issues to show merge progress
-	for _, path := range intermediateProject.Include {
+	if len(intermediateProject.Include) > 0 {
 		if opts == nil {
 			err = errors.New("trying to open include files with empty options")
 			return nil, errors.Wrapf(err, LoadProjectError)
 		}
-		opts.UpdateForFile(path.FileName)
-
-		var yaml []byte
-		opts.Identifier = identifier
-		opts.RemotePath = path.FileName
-		grip.Debug(message.Fields{
-			"message":     "retrieving included YAML file",
-			"remote_path": opts.RemotePath,
-			"read_from":   opts.ReadFileFrom,
-			"module":      path.Module,
-		})
-		if path.Module != "" {
-			yaml, err = retrieveFileForModule(ctx, *opts, intermediateProject.Modules, path.Module)
-		} else {
-			yaml, err = retrieveFile(ctx, *opts)
+		wg := sync.WaitGroup{}
+		yamlChan := make(chan yamlTuple, len(intermediateProject.Include))
+		includeChan := make(chan Include)
+
+		// Be polite. Don't make more than 10 concurrent requests to GitHub.
+		for i := 0; i < 10; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for include := range includeChan {
+					processIntermediateProjectInludes(ctx, identifier, intermediateProject, include, yamlChan, opts)
+				}
+			}()
 		}
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: retrieving file '%s'", LoadProjectError, path.FileName)
+
+		for _, path := range intermediateProject.Include {
+			includeChan <- path
 		}
-		add, err := createIntermediateProject(yaml, opts.UnmarshalStrict)
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: loading file '%s'", LoadProjectError, path.FileName)
+
+		close(includeChan)
+		wg.Wait()
+		close(yamlChan)
+
+		// We must apply the includes in order, so we indirect through a map.
```

Comment:
I'm not sure that this comment actually helps with understanding. The reason the includes can be applied in order is because of the combination of `wg.Wait` waiting for all includes to be loaded into `yamlChan`, then iterating in order on L744 through `intermediateProjet.Include` again.

I would recommend either remove or rephrase + move down.